modules/vpc/main.tf

```diff
@@ -81,7 +81,7 @@ data "aws_availability_zones" "current" {
 # ------------------------------------------------------------
 
 locals {
-  num_availability_zones = var.num_availability_zones == null ? length(data.aws_availability_zones.current) : var.num_availability_zones
+  num_availability_zones = var.num_availability_zones == null ? length(data.aws_availability_zones.current.names) : var.num_availability_zones
```

Comment:
If we have to reference `.names` here. Does this also need to be referenced later in the config where the actual availability zone name is being used?